File: lavanderia.pro/api/types/product.go
package types

import "time"

type Product struct {
	ID        string     `json:"id" bson:"_id,omitempty" uri:"id"`
	Name      string     `json:"name"`
	Price     float64    `json:"price"`
	Business  string     `json:"business,omitempty" bson:"business,omitempty"`
	CreatedAt *time.Time `json:"created_at,omitempty" bson:"created_at"`
	UpdatedAt *time.Time `json:"updated_at,omitempty" bson:"updated_at"`
	DeletedAt *time.Time `json:"deleted_at,omitempty" bson:"deleted_at"`
}

File: app/lib/models/product.dart
class Product {
  final String? id;
  final String? name;
  final num? price;
  final String? createdAt;
  final String? updatedAt;

  Product({this.id, this.name, this.price, this.createdAt, this.updatedAt});

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      price: json['price'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }
}